modules/labs-react/text-input/lib/useTextInputModel.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
+import {
+  createEventMap,
+  ErrorType,
+  Model,
+  ToModelConfig,
+  useEventMap,
+  useUniqueId,
+} from '@workday/canvas-kit-react/common';
+
+type TextInputState = {
+  hasError?: ErrorType;
+  inputId?: string;
+  hintId?: string;
+};
+
+type TextInputEvents = {
+  addError(): void;
+  removeError(): void;
+  addAlert(): void;
+  removeAlert(): void;
```

Comment:
Thinking about this more I wonder if Alert even really needs to be a built in state. It only controls the error ring color so you could really just manually add that yourself around TextInput.Field. In that case we could simplify, make hasError a boolean. And then just add example of how to set the color to anything, Alert Yellow, Green success, etc. Thoughts?

Reply:
Typically in form libraries or form usage, the error is a property of the form model and not each field. It is a prop. If you add something like `hasError` or `errorType` to `TextInputModelConfig`, it could be simply forwarded to the state. Using events adds complexity to integrating with libraries. I can see it in the example:

https://github.com/Workday/canvas-kit/pull/1201/files#diff-8bfe0cebce955a2e1f5d234f29843603d8a03b04e376bdad11bddf10aec11d12R52-R63

Reply:
Is this the right way to forward the state?
https://codesandbox.io/s/hopeful-kalam-x5cji?file=/tester/Tester.tsx:730-767

Reply:
No. It would be more like:

```tsx
const useTextInputModel = ({hasError}) => {

  const state = {
    hasError,
  }

  const events = {}
  
  return { state, events }
}
```